app/models/concerns/thredded/topic_common.rb

```diff
@@ -47,22 +47,30 @@ def unread(user)
           .merge(reads_class.where(reads[:id].eq(nil).or(reads[:read_at].lt(topics[:updated_at]))))
       end
 
+      def read_states_by_topics_lookup(user)
```

Comment:
Complex methods should have documented return type and arguments. Also, looks like this one should be private?